.github/scripts/nuget-install-wdk.ps1

```diff
@@ -0,0 +1,121 @@
+param (
+    [string]$wdkVersion
+)
+
+# Extract the first three parts of the version
+$majorVersionNum = ($wdkVersion -split '\.')[0..2] -join '.'
+
+Write-Host "Installing WDK version $wdkVersion"
+Write-Host "Major version number: $majorVersionNum"
+
+# Update the packages.config file with the version passed as parameter
+$packagesConfigPath = ".\packages.config"
+
+if (Test-Path $packagesConfigPath) {
+    [xml]$packagesConfig = Get-Content $packagesConfigPath
+
+    foreach ($package in $packagesConfig.packages.package) {
+        if ($package.id -like "Microsoft.Windows.*") {
+            $package.version = $wdkVersion
+        }
+    }
+
+    $packagesConfig.Save($packagesConfigPath)
+    Write-Host "Updated packages.config with version $wdkVersion"
+} else {
+    Write-Error "packages.config file not found"
+}
+
+# Install WDK using NuGet
+try {
+    nuget restore .\packages.config -PackagesDirectory C:\WDK
+    Write-Host "WDK installed at C:\WDK"
+} catch {
+    Write-Error "Failed to restore packages using NuGet. Error: $_"
+    exit 1
+}
+
+$folders = @(
```

Comment:
why is there all this manual copying of folders after the nuget packages are restored? this seems very brittle

Reply:
The `setup_path` function sets the `bin` and `tools` root W.R.T standard WDK's directory structure. However, with NuGet I noticed that some bins and headers were placed in a different path. So, this script moves them under the root that `setup_path` is using.

https://github.com/microsoft/windows-drivers-rs/blob/10a8e3742b7834e816f6a2ce630b46533bcec357/crates/wdk-build/src/cargo_make.rs#L515

Reply:
this function actually uses `detect_wdk_content_root`, so as long as that's detecting nuget properly, then we're all set. 

see https://github.com/microsoft/windows-drivers-rs/pull/236#issuecomment-2518603876 for more details, but the tl;dr is that the nuget WDK's use in c drivers works by setting the include and bin paths via modifying the vcxproj. as stated in my linked comment, since there is no path forward to use the same solution that nuget uses in c drivers (ie. vcxproj is not supported by our rust tooling), we should rely on the env vars used by ewdk